testing/endtoend/evaluators/fee_recipient.go

```diff
@@ -0,0 +1,140 @@
+	dirs, err := ioutil.ReadDir(filepath.Clean(testNetDir))
+	if err != nil {
+		return err
+	}
+	for _, f := range dirs {
+		if f.IsDir() {
+			configPath := filepath.Join(testNetDir, filepath.Clean(f.Name()+"/config.json"))
+			jsonFile, err := os.Open(filepath.Clean(configPath))
+			if err != nil {
+				return err
+			}
+			var configFile validator_service_config.ProposerSettingsPayload
+			if err := json.NewDecoder(jsonFile).Decode(&configFile); err != nil {
+				return err
+			}
+			configFiles = append(configFiles, &configFile)
+		}
+	}
+
+	for _, ctr := range blks.BlockContainers {
+		switch ctr.Block.(type) {
+		case *ethpb.BeaconBlockContainer_BellatrixBlock:
+			var account common.Address
+
+			fr := ctr.GetBellatrixBlock().Block.Body.ExecutionPayload.FeeRecipient
+			if len(fr) != 0 && hexutil.Encode(fr) != params.BeaconConfig().EthBurnAddressHex {
+				account = common.BytesToAddress(fr)
+			} else {
+				return errors.New("fee recipient is not set")
+			}
+			validatorRequest := &ethpb.GetValidatorRequest{
+				QueryFilter: &ethpb.GetValidatorRequest_Index{
+					Index: ctr.GetBellatrixBlock().Block.ProposerIndex,
+				},
+			}
+			validator, err := client.GetValidator(context.Background(), validatorRequest)
+			if err != nil {
+				return errors.Wrap(err, "failed to get validators")
+			}
+			publickey := validator.GetPublicKey()
+			usesDefaultFeeRecipient := false
+			for _, config := range configFiles {
+				option, ok := config.ProposerConfig[hexutil.Encode(publickey)]
+				if ok {
+					if option.FeeRecipient != account.Hex() {
+						return fmt.Errorf("fee recipient %s does not match the proposer settings fee recipient %s", account.Hex(), option.FeeRecipient)
+					}
+					usesDefaultFeeRecipient = true
+				}
+			}
+			if !usesDefaultFeeRecipient {
+				if configFiles[0].DefaultConfig.FeeRecipient != account.Hex() {
+					return fmt.Errorf("fee recipient %s does not match the default fee recipient %s", account.Hex(), configFiles[0].DefaultConfig.FeeRecipient)
+				}
+			}
+			currentBlock, err := web3.BlockByHash(ctx, common.BytesToHash(ctr.GetBellatrixBlock().GetBlock().GetBody().GetExecutionPayload().BlockHash))
+			if err != nil {
+				return err
+			}
+
+			accountBalance, err := web3.BalanceAt(ctx, account, currentBlock.Number())
+			if err != nil {
+				return err
+			}
+			previousBlock, err := web3.BlockByHash(ctx, common.BytesToHash(ctr.GetBellatrixBlock().GetBlock().GetBody().GetExecutionPayload().ParentHash))
+			if err != nil {
+				return err
+			}
+			prevAccountBalance, err := web3.BalanceAt(ctx, account, previousBlock.Number())
+			if err != nil {
+				return err
+			}
+			if currentBlock.GasUsed() > 0 && accountBalance.Uint64() <= prevAccountBalance.Uint64() {
+				log.Infof("current block num: %d , previous block num: %d , account balance: %d,  pre account balance %d", currentBlock.Number(), previousBlock.Number(), accountBalance, prevAccountBalance)
```

Comment:
do we need this log ?

Reply:
this log helped me realize it was at 0 , ideally this error isn't thrown in the first place but it did help me when the error was thrown

Reply:
Yeah, that sounds fine. But I am assuming, now that you know it works correctly you don't need it for debugging purposes, correct ?